File: leo/week9/6603.js
// Run by Node.js
const readline = require("readline");

(async () => {
  const rl = readline.createInterface({ input: process.stdin });
  const input = [];

  for await (const line of rl) {
    input.push(line.trim());
  }

  rl.close();

  const generateCombinations = (arr, start, combination, results) => {
    if (combination.length === 6) {
      results.push([...combination]);
      return;
    }

    for (let i = start; i < arr.length; i++) {
      combination.push(arr[i]);
      generateCombinations(arr, i + 1, combination, results);
      combination.pop();
    }
  };

  let idx = 0;
  while (idx < input.length) {
    const line = input[idx];
    if (line === "0") break;

    const parts = line.split(" ").map(Number);
    const k = parts[0];
    const S = parts.slice(1);

    const results = [];
    generateCombinations(S, 0, [], results);

    results.forEach((seq) => {
      console.log(seq.join(" "));
    });

    console.log("");
    idx++;
  }
})();

File: ella/week9/6603.py
import sys
input = sys.stdin.readline

def back(a1, a2):
    if len(a2) == 6:
        print(" ".join(map(str, a2)))
        return
    else:
        for i in a1:
            if len(a2) == 0 or i > a2[-1]:
                a2.append(i)
                back(a1, a2)
                a2.pop()

while True:
    arr = list(map(int, input().split()))
    k = arr[0]
    if k == 0:
        break
    s = arr[1:]
    answer = []

    back(s, answer)
    print()